nonce.go

```diff
@@ -131,16 +150,40 @@ func (n Nonce) Validate() error {
 	return isValidNonce(n.value)
 }
 
-// MarshalJSON encodes the receiver Nonces as a JSON string
+// MarshalJSON encodes the receiver Nonces as either a JSON string (if the array
+// comprises only one element) or as an array of JSON strings.
 func (ns Nonces) MarshalJSON() ([]byte, error) {
 	if len(ns) == 1 {
-		return json.Marshal(ns[0].value)
+		return json.Marshal(ns[0])
 	}
 
-	return nil, errors.New("TODO handle array of nonces")
+	return json.Marshal([]Nonce(ns))
 }
 
+// UnmarshalJSON decodes the EAT nonce in JSON format
 func (ns *Nonces) UnmarshalJSON(data []byte) error {
+	if isJSONArray(data) {
+		return json.Unmarshal(data, (*[]Nonce)(ns))
+	}
+
+	var n Nonce
+
+	if err := json.Unmarshal(data, &n); err != nil {
+		return err
+	}
+
+	*ns = Nonces{n}
+
+	return nil
+}
+
+// MarshalJSON encodes the receiver (non-array) nonce as a JSON string
+func (n Nonce) MarshalJSON() ([]byte, error) {
```

Comment:
A general comment. Nonce and Nonces are handled as different types effectively and have different MarshalJSON and UnmarshalJSON method, however the name type (Nonce vs Nonces) are strikingly similar!. This could be a little puzzling to the reader. Consider renaming Nonces to a suitable name which could clarify things a bit ?

Reply:
I think the striking similarity is a desired feature: `Nonces` is the "main" type, mapping the EAT spec.  `Nonce` is a sugar type for the (very common case) where there's only one nonce.

Reply:
As discussed, one possible approach to improve naming is to have a possible approach below:

1) Change Nonces to NonceList or NonceClaim

// Append a nonce to a list
// This needs to have a check for valid Nonce, 
2 Append( ):

Reply:
while at ASDA I have been pondering on this and scribbled this revised interface:
```
package eat

type Token struct {
	Nonce eat.Nonce
}

func encode() (*Token, error) {
	nonce := eat.Nonce{}

	if err := nonce.Add("deadbeef"); err != nil {
		return nil, err
	}

	// if needed add more nonces
	if err := nonce.Add("beefbaaf"); err != nil {
		return nil, err
	}

	return &Token{
		Nonce: nonce,
	}, nil
}

func decode(data []byte) error {
	t Token{}
	
	if err := t.UnmarshalCBOR(data); err != nil {
		return err
	}

	nonceBytes []byte

	n := t.Nonce.Len()

	for i := 0; i < n; i++ {
		nonceBytes = t.Nonce.GetI(i)

		// do something with nonce
	}
}
```

WDYT?

Reply:
Concept is fine with me. A Unified interface is the preferred option. Go for it, happy to review complete change!